lnrpc/autopilotrpc/autopilot_server.go

```diff
@@ -85,36 +91,42 @@ var _ AutopilotServer = (*Server)(nil)
 // this method. If the macaroons we need aren't found in the filepath, then
 // we'll create them on start up. If we're unable to locate, or create the
 // macaroons we need, then we'll return with an error.
-func New(cfg *Config) (*Server, lnrpc.MacaroonPerms, error) {
+func New() (*Server, lnrpc.MacaroonPerms, error) {
 	// We don't create any new macaroons for this subserver, instead reuse
 	// existing onchain/offchain permissions.
-	server := &Server{
-		cfg:     cfg,
-		manager: cfg.Manager,
-	}
-
-	return server, macPermissions, nil
+	return &Server{cfg: &Config{}}, macPermissions, nil
 }
 
-// Start launches any helper goroutines required for the Server to function.
+// Start normally launches any helper goroutines required for the Server to
+// function, but in this case, it is a noop as the autopilot manager is started
+// once the dependencies have been injected.
 //
 // NOTE: This is part of the lnrpc.SubServer interface.
 func (s *Server) Start() error {
-	if atomic.AddInt32(&s.started, 1) != 1 {
-		return nil
-	}
-
-	return s.manager.Start()
+	// The autopilot manager cannot be started until we have the
```

Comment:
Yeah good point! I removed the `Start` functions in all of the RPC sub-servers.
After doing so, the main RPC server's `Start` function also became pointless as it no longer executed any code, so I also removed that.

Removal of the main RPC server's `Start` function felt a bit more incorrect though... But the current code does motivate that removal.

If we do not like the removal main RPC server's `Start` function, I would like to flag again that I think the other alternative would be an option that we tried to avoid previously:
We could always rethink and remove this "`InjectDependencies`" pattern, and potentially create and start the `WalletKit` RPC server before any of the other sub-servers are created. That will be unclean in other aspects though, so I'm not sure if that'd be preferable...